File: src/agentcap/followups/continue_.py
"""Literal-``continue`` follow-up strategy."""

from __future__ import annotations

from . import FollowUp


class ContinueFollowUp(FollowUp):
    name = "continue"

    def __init__(self, text: str = "continue") -> None:
        self.text = text

    def next(self, *, original_task: str, last_response: str, turn: int) -> str:
        return self.text

File: src/followups/continue_.rs
//! Literal-`continue` follow-up.

use super::FollowUp;

pub struct ContinueFollowUp {
    text: String,
}

impl Default for ContinueFollowUp {
    fn default() -> Self {
        ContinueFollowUp {
            text: "continue".to_string(),
        }
    }
}

impl FollowUp for ContinueFollowUp {
    fn next(&self, _original_task: &str, _last_response: &str, _turn: i64) -> String {
        self.text.clone()
    }
}
